docs/design/w3c/w3c-representation.md

```diff
@@ -0,0 +1,786 @@
+        * For example: the length of encoded credential signature string is about 2.5 times less than in the basic approach
+        * Find an [example data](./encoding.md#example) to see the difference
+    * **Answer**:
+        * Start from basic Aries encoding.
+        * Experiment with other algorithms (CBOR, Message Pack, Protobuf, Custom) as a next step.
+* Q6: W3C data model allows attributes to be represented not only as key/value string pairs but also as
+  objects and arrays.
+    * If we want to support more complex representations for the W3C AnonCreds credential attributes and their
+      presentations, we need to design following things:
+        * how encoding will work for such attributes
+        * how selective disclosure will work on top level attribute itself
+            ```
+              "credentialSubject": {
+                "address": {
+                    "type": "Address",
+                    "city": "Foo",
+                    "street": "Main str."
+                }
+              }
+            ```
+    * **Answer**:
+        * Only key/value string pairs will be supported for the current phase
+* Q7: Should we care about back way conversion of credential issued in W3C form?
+    * Assume that we issued W3C which cannot convert back into legacy form
+    * For example supporting different attributes types can be used in credentialSubject (support nested objects, array,
+      other features)
+        * Need to decide on object encoding algorithm
+    * **Answer**: Yes. Within the current design/effort credentials must be convertible back and forth
+* Q8: Should we include Holder DID into credential subject as [`id` attribute](https://www.w3.org/TR/vc-data-model/#identifiers)?
+    * This enables credential subject [validation](https://www.w3.org/TR/vc-data-model/#credential-subject-0)
+    * We can add if for newly issued credentials but cannot set during the conversion of previously issued credentials.
+    * **Answer**:
+        * Up to Issuer, Optionally Issuer may include DID into credential subject
+        * anoncreds-rs will provide a helper function to set `id`
+* Q9: Predicates representation in credential subject
+    * Derive an attribute and put it into `credentialSubject` like it demonstrated
+      in the [specification](https://www.w3.org/TR/vc-data-model/#presentations-using-derived-credentials)
+        * Example:
+            * For Predicate: `{"name": "birthdate", "p_type": "<", "value":"20041012"}`
+            * Derived attribute: `{"birthdate": "birthdate less 20041012"}`
+        * During the `proofValue` crypto verification we can parse the phrase and restore predicate attributes
+    * Put predicate as object into `credentialSubject`
+        ```
+          "credentialSubject": {
+            ...
+            "birthdate": {
+              "type": "Predicate",
+              "p_type": "<", 
+              "value": "20041012"
+            }
```

Comment:
I think this syntax could work, but implementations should handle an array value here in the case of multiple predicates on the same attribute. Or both a value reveal and a predicate on the same attribute, for example:

```json
            "birthdate": [
              "20040101",
              {
                "type": "Predicate",
                "p_type": "<", 
                "value": "20041012"
              }
            ],
```

Reply:
This case makes sense: `multiple predicates on the same attribute`
But this one is not valid and will fail on the `cl` library level: `both a value reveal and a predicate on the same attribute`